Sources/PostgresNIO/Connection/PostgresConnection+Authenticate.swift

```diff
@@ -60,12 +63,64 @@ private final class PostgresAuthenticationRequest: PostgresRequest {
                     return try [PostgresMessage.Password(string: hash).message()]
                 case .plaintext:
                     return try [PostgresMessage.Password(string: self.password ?? "").message()]
+                case .saslMechanisms(let saslMechanisms):
+                    if saslMechanisms.contains("SCRAM-SHA-256") && self.password != nil {
+                        let saslManager = SASLAuthenticationManager(asClientSpeaking:
+                            SASLMechanism_SCRAM_SHA256(username: self.username, password: { self.password! }))
```

Comment:
If this is all handled implicitly, does `SASLMechanism_SCRAM_SHA256` and friends need to be public?

Reply:
It was originally intended for these to be more globally useful, not limited to Postgres. I suppose they should be kept internal until such time as they actually _are_ ready for such use though.